local_ads/campaign_serialization.cpp

```diff
@@ -11,6 +11,15 @@
 
 namespace
 {
+using namespace local_ads;
+
+auto const kHalfByteShift = 0x4;
```

Comment:
Please don't use hex notation for kHalfByteShift cause it creates an impression that bit structure of the value matters. You are not going to use it in bit-wize and expressions.